src/utils/todo.ts

```diff
@@ -23,6 +26,190 @@ export const buildTodoItemIfFromAffiliationInvitation =
     return newTodo
   }
 
+export const buildTodoItemFromTasks = (task: TaskI) : TodoItemI | null => {
+  if (task.task.todo) {
+    return buildTodo(task)
+  } else if (task.task.filing) {
+    return buildFilinngTodo(task)
+  } else {
+    console.error('ERROR - got unknown task =', task)
+    return null
+  }
+}
+
+const buildTodo = (task: TaskI) : TodoItemI | null => {
+  const todo = task.task.todo
+  const header = todo.header
+  let newTodo: TodoItemI | null = null
+
+  if (header) {
+    switch (header.name) {
+      case FilingTypes.ANNUAL_REPORT:
+        newTodo = loadAnnualReportTodo(task)
+        break
+      case FilingTypes.CONVERSION:
+        /** TO-DO: 'Conversion' Action */
+        break
+      default:
+        console.error('ERROR - invalid name in todo header =', header)
+        break
+    }
+  } else {
+    console.error('ERROR - invalid header in todo =', todo)
+  }
+  return newTodo
+}
+
+/** Loads a NEW Annual Report todo. */
+const loadAnnualReportTodo = (task: TaskI) : TodoItemI | null => {
+  const todo = task.task.todo
+  const header = todo.header
+  const business = useBcrosBusiness()
+  if (business && header) {
+    const ARFilingYear = header.ARFilingYear
+
+    const subtitle = (task.enabled && !business.isBaseCompany())
+      ? '(including Address and/or Director Change)'
+      : null
+
+    const enabled = task.enabled && !business.isDisableNonBenCorps()
+
+    // TO-DO: business.isAllowed() will always return false for KIAL DEV 1 account
```

Comment:
If not already created, please create a ticket for this issue so we look into it.